server/core/src/indexer/fulltextPush.ts

```diff
@@ -95,6 +97,39 @@ export class FullTextPushStage implements FullTextPipelineStage {
     return { docs: [], pass: true }
   }
 
+  async indexRefAttributes (
+    attributes: Map<string, AnyAttribute>,
+    doc: DocIndexState,
+    elasticDoc: IndexedDoc,
+    metrics: MeasureContext
+  ): Promise<void> {
+    for (const attribute in doc.attributes) {
+      const { attr } = extractDocKey(attribute)
+      const attrObj = attributes.get(attr)
+      if (
+        attrObj !== null &&
+        attrObj !== undefined &&
+        attrObj.index === IndexKind.FullText &&
+        attrObj.type._class === core.class.RefTo
+      ) {
+        const refs = doc.attributes[attribute].split(',')
```

Comment:
Why id attribute is one ref, if you want array too it should check also ArrOf RefTo

Reply:
Added support for ArrOf